quinn-proto/src/connection/streams.rs

```diff
@@ -321,6 +321,24 @@ impl Streams {
         stream_frames
     }
 
+    /// Notify the application that new streams were opened or a stream became readable.
+    pub fn on_stream_frame(&mut self, notify_readable: bool, stream: StreamId) {
+        if stream.initiator() == self.side {
```

Comment:
This condition (which also appears in the old version) appears to be wrong on the face of it: wouldn't `stream.initiator() == self.side` be exactly about opening locally-initated streams?

Reply:
The purpose of this method is to prepare events reflecting streams becoming opening and readable. We don't want to emit events for the opening of locally-initiated streams, so we skip that check here. The logic/naming/comments could probably be a lot clearer, e.g. by just adding `stream.initiator() != side &&` to the next if statement.